using System;
using System.Runtime.Serialization;

namespace MalwareScan.Client
{
	/// <summary>
	/// Represents an error that occurred because a duplicate evaluation was found.
	/// </summary>
	public class DuplicateEvaluationFoundException : DuplicateObjectFoundException
	{
		/// <summary>
		/// Initializes a new instance of the <see cref="DuplicateEvaluationFoundException"/> class.
		/// </summary>
		public DuplicateEvaluationFoundException()
		{
		}

		/// <summary>
		/// Initializes a new instance of the <see cref="DuplicateEvaluationFoundException"/> class with a specified error message.
		/// </summary>
		/// <param name="message">the message that describes the error</param>
		public DuplicateEvaluationFoundException(string message)
			: base(message)
		{
		}

		/// <summary>
		/// Initializes a new instance of the <see cref="DuplicateEvaluationFoundException"/> class with a specified error message and
		/// a reference to the inner exception that is the cause of this exception.
		/// </summary>
		/// <param name="message">the message that describes the error</param>
		/// <param name="innerException">the exception that is the cause of the current exception, or a null reference if no inner
		/// exception is specified</param>
		public DuplicateEvaluationFoundException(string message, Exception innerException)
			: base(message, innerException)
		{
		}

		/// <summary>
		/// Initializes a new instance of the <see cref="DuplicateEvaluationFoundException"/> class with serialized data.
		/// </summary>
		/// <param name="info">the <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown
		/// </param>
		/// <param name="context">the <see cref="StreamingContext"/> that contains contextual information about the source or destination
		/// </param>
		protected DuplicateEvaluationFoundException(SerializationInfo info, StreamingContext context)
			: base(info, context)
		{
		}
	}
}
